test/filter_test.cc

```diff
@@ -75,7 +76,7 @@ int main(int argc, char* argv[]) {
 
   // Make sure we ran all of the tests
   const size_t count = test_reporter.GetCount();
-  const size_t expected = (argc == 2) ? std::stoul(argv[1]) : count;
+  const size_t expected = (argc == 2) ? std::atol(argv[1]) : count;
```

Comment:
If `stringstream` works on Android, I vote for using that.  It's a bit more code, but at least it will be clear what happens and we won't have a signed->unsigned cast hidden in the middle of a complicated expression.